### DISCUSSION TITLE
How to change the compression level when using the `--onefile` option?

### DISCUSSION THREAD

**Discussion by 34j** [pyinstaller/pyinstaller#7908] [Help] [answered]
My project includes PyTorch for CUDA, and when I use the `--onefile` option, it takes about 30 seconds to decompress to a temporary directory, even with an NVMe SSD, and the startup is a bit slow. (When using the `--onefile` option, the overall file/folder size is reduced to 60%, so it appears that some compression is taking place.) As it seems using uncompressed ZIP is somewhat faster compared to the normal compression level when zipping and unzipping the output files (generated without using --onefile option), I would like to use uncompressed compression method (?) when using the `--onefile` option, but is this impossible? (Sorry if my English is terrible.)

**Comment by rokm** [comment, marked as answer]
No, there is no uncompressed option for `onefile` builds. The onefile mode was never intended for large applications that use PyTorch and CUDA, anyway - if you find it too slow, look for a different solution (or modify the PyInstaller code to suit your requirements).

**Reply by 34j** [reply]
How easy would it be to modify the PyInstaller code?

**Reply by rokm** [reply]
> How easy would it be to modify the PyInstaller code?

Try and see how it goes? I have no way of gauging your skill level...
